### PR TITLE
Add zero address checks

### PR DESCRIPTION
**Pull request by hswopeams**

### CODE REVIEW

File: contracts/VoucherKernel.sol [hunk 1]
```diff
@@ -172,11 +172,19 @@ contract VoucherKernel is IVoucherKernel, Ownable, Pausable, UsingHelpers {
...
+
     /**
      * @notice Construct and initialze the contract. Iniialises associated contract addresses, the complain period, and the cancel or fault period
      * @param _tokensContract address of the associated ERC1155ERC721 contract instance
      */
-    constructor(address _tokensContract) {
+    constructor(address _tokensContract)
+    notZeroAddress(_tokensContract)
+    {
+        require(_tokensContract != address(0), "0A");
```
Comment (bidyut-arianelabs) [inline contracts/VoucherKernel.sol:187]: Shouldn't this line be removed now that we have `notZeroAddress` modifier?

**Review by bidyut-arianelabs — CHANGES_REQUESTED** [review]

File: contracts/VoucherKernel.sol (hunk 1 above)
Comment (hswopeams) [reply contracts/VoucherKernel.sol:187]: Yep. I forgot to remove it. I'll remove it in the next commit.

**Review by bidyut-arianelabs — DISMISSED** [review]
Looks good to me 👍

**Review by zajck — COMMENTED** [review]
Generally it looks ok, but I have one question before approving:

In some functions (e.g. `Cashier._withdrawPayments`, `Cashier._withdrawDeposits` and some other places) , we do check `require(_recipient != address(0), "UNSPECIFIED_ADDRESS");` which is basically the same as it would be done with notZeroAddress modifier. I know that this is there already from before, but I just wonder if there is any specific reason not to use the modifier?

File: contracts/Cashier.sol [hunk 2]
```diff
@@ -815,7 +819,9 @@ contract Cashier is ICashier, UsingHelpers, ReentrancyGuard, Ownable, Pausable {
         uint256 _amount,
         uint8 _paymentMethod,
         uint256 _tokenIdSupply
-    ) internal {
+    ) internal    
+      notZeroAddress(_recipient)
+    {
         require(_recipient != address(0), "UNSPECIFIED_ADDRESS");
```
Comment (zajck) [inline contracts/Cashier.sol:825]: `require` can now be removed`require` can now be removed

**Review by zajck — CHANGES_REQUESTED** [review]
Beside what I put the in code comments - there are other places that we use two different checks for `nonZeroAddress` (in some functions with modifiers, in others with explicit require statements). This happens in `Cashier`, `Gate` and `VoucherKernel`. But changing this might even be a task for separate PR.
We can make it a separate issue and I'll aprove this after you remove those two lines I commented.

**Review by bidyut-arianelabs — APPROVED** [review]
Looks good to me 👍

**Review by zajck — APPROVED** [review]

**Pull request merged**